src/main/java/uk/gov/hmcts/divorce/common/event/SwitchedToSoleCo.java

```diff
@@ -115,51 +93,30 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
             switchToSoleService.switchApplicantData(data);
         }
 
-        generateSwitchToSoleDocuments(details, data, caseId);
-
-        var state = details.getState() == JSAwaitingLA ? JSAwaitingLA : AwaitingLegalAdvisorReferral;
-
-        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-            .data(data)
-            .state(state)
-            .build();
-    }
-
-    private void generateSwitchToSoleDocuments(CaseDetails<CaseData, State> details,
-                                               CaseData caseData,
-                                               Long caseId) {
-        if ((CO_D84.equals(caseData.getDocuments().getTypeOfDocumentAttached())
-                || D84.equals(caseData.getDocuments().getScannedSubtypeReceived()))
-            && SWITCH_TO_SOLE.equals(caseData.getConditionalOrder().getD84ApplicationType())) {
-
-            if (caseData.isJudicialSeparationCase()) {
-                if (caseData.getApplicant2().isRepresented()) {
-                    generateJudicialSeparationSwitchToSoleSolicitorLetter.apply(caseData, caseId, caseData.getApplicant1(),
-                            caseData.getApplicant2());
-                } else {
-                    generateSwitchToSoleJSLetter.apply(caseData, caseId, caseData.getApplicant1(), caseData.getApplicant2());
-                }
-            } else {
-                generateSwitchToSoleCoLetter.apply(caseData, caseId, caseData.getApplicant1(), caseData.getApplicant2());
-            }
-        }
-    }
-
-    public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
-                                               CaseDetails<CaseData, State> beforeDetails) {
+        final var state = details.getState() == JSAwaitingLA ? JSAwaitingLA : AwaitingLegalAdvisorReferral;
 
         log.info("SwitchedToSoleCO submitted callback invoked for case id: {}", details.getId());
 
-        final CaseData data = details.getData();
         notificationDispatcher.send(switchToSoleCoNotification, data, details.getId());
 
         if (CO_D84.equals(data.getDocuments().getTypeOfDocumentAttached())
-                || D84.equals(data.getDocuments().getScannedSubtypeReceived())
+            || D84.equals(data.getDocuments().getScannedSubtypeReceived())
             && SWITCH_TO_SOLE.equals(data.getConditionalOrder().getD84ApplicationType())) {
 
-            switchToSoleCoPrinter.print(data, details.getId(), data.getApplicant2());
+            var documentPackInfo =
+                switchToSoleConditionalOrderDocumentPack.getDocumentPack(data, null);
```

Comment:
Why isn't this call in the notification class?

Reply:
I can move it but as the original was placed here, I left it here.

Reply:
oh right as in it would send an email AND send to bulk print?

Reply:
kinda, itll send the email yh but the bulk print will depend on the condition as its in an if statement.

Reply:
cool well as long as it works we can leave as is